### ISSUE TITLE
Failed to read provider schema

### ISSUE THREAD

**Issue by ming-raf** [cloudflare/cf-terraforming#553]
### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the library and it is still present.

### cf-terraforming version

v0.12.0

### Expected outcome

When running cf-terraforming 'generate' it should work normally.

### Actual outcome

Running on Windows 11 with cf-terraforming v0.12.0

I am bit new with Terraform but it does not seem anything is configured incorrectly with `terraform init`
main.tf:
```
terraform {
  required_providers {
    cloudflare = {
      source = "cloudflare/cloudflare"
    }
  }
}

provider "cloudflare" {
  api_token = "<API-TOKEN>"
}
```
```
# terraform init

Initializing the backend...

Initializing provider plugins...
- Finding latest version of cloudflare/cloudflare...
- Installing cloudflare/cloudflare v4.3.0...
- Installed cloudflare/cloudflare v4.3.0 (self-signed, key ID C76001609EE3B136)

Partner and community providers are signed by their developers.
If you'd like to know more about provider signing, you can read about it here:
https://www.terraform.io/docs/cli/plugins/signing.html

Terraform has created a lock file .terraform.lock.hcl to record the provider
selections it made above. Include this file in your version control repository
so that Terraform can guarantee to make the same selections by default when
you run "terraform init" in the future.
```

When running cf-terraforming:
```
$ cf-terraforming.exe -a <ACCOUNT-ID> -v generate --resource-type dns
time="2023-04-05T10:27:13+10:00" level=debug msg="initializing cloudflare-go with API Token" account_Id=<ACCOUNT-ID> zone_id=
time="2023-04-05T10:27:15+10:00" level=debug msg="initializing Terraform in ."
time="2023-04-05T10:27:15+10:00" level=debug msg="reading Terraform schema for Cloudflare provider"
time="2023-04-05T10:27:16+10:00" level=fatal msg="failed to read provider schemaexit status 1\n\nError: Required plugins are not installed\n\nThe installed provider plugins are not consistent with the packages selected\nin the dependency lock file:\n  - registry.terraform.io/cloudflare/cloudflare: there is no package for registry.terraform.io/cloudflare/cloudflare 4.3.0 cached in .terraform\\providers\n\nTerraform uses external plugins to integrate with a variety of different\ninfrastructure services. You must install the required plugins before running\nTerraform operations.\n\n"
```

### Steps to reproduce

1. mkdir a-new-folder
2. cd a-new-folder
3. create main.tf with with cloudflare provider
4. run `terraform init`
5. run `cf-terraforming generate`

**Comment by willswire** [comment]
Also receiving the same error

**Comment by ming-raf** [comment]
Sorry didnt mean to close

**Comment by ming-raf** [comment]
Seems like an issue with Windows builds. Been using it fine on WSL/Ubuntu.

**Comment by pnwstevan** [comment]
Confirmed - exact same issue and fix as @christiangonre - arm64 binary throws this error, amd64 binary from release works fine; also on M1.

**Comment by raisin-loaf** [comment]
I used `cf-terraforming` successfully in September of last year. Trying again today, with v0.12, I ran into the same problem as the OP. I tested back sequentially and found the problem was introduced with v0.9. Reading the change notes this [PR](https://github.com/cloudflare/cf-terraforming/pull/488) stood out (I had no idea `cf-terraforming` performs a Terraform installation by default). Sure enough, using the `--teraform-binary-path` option and pointing it at my local binary solved the problem.
